### PR TITLE
Add Wikicorpus dataset

### PR DESCRIPTION
**Pull request by albertvillanova**
Add dataset.

### CODE REVIEW

File: datasets/wikicorpus/README.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
+---
+annotations_creators:
+  raw_ca: []
+  raw_en: []
+  raw_es: []
```
Comment (yjernite) [inline datasets/wikicorpus/README.md:5]: could you fix these? There's a `no-annotation` option in the tagging app

Comment (albertvillanova) [reply datasets/wikicorpus/README.md]: Done.

File: datasets/wikicorpus/wikicorpus.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "dirpath": downloaded_dir,
+                },
+            ),
+        ]
+
+    def _generate_examples(self, dirpath):
+        for filepath in Path(dirpath).iterdir():
```
Comment (lhoestq) [inline datasets/wikicorpus/wikicorpus.py:125]:
```suggestion
        for filepath in sorted(Path(dirpath).iterdir()):
```
we need to make this deterministic

**Review by lhoestq — COMMENTED** [review]
Looks really good thank !

Could to try to reduce the size of the dummy_data.zip files ? The current one are quite heavy (500KB for the english one for example).

To do so you can for example take a look inside the dummy_data.zip file of the `en` config  and remove unnecessary files to only keep 1 or 2.

**Comment by albertvillanova** [comment]
@lhoestq done! ;)

File: datasets/wikicorpus/wikicorpus.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                    "title": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                }
+            )
+        elif self.config.form == "tagged":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "title": datasets.Value("string"),
+                    "sentence": datasets.Sequence(datasets.Value("string")),
```
Comment (lhoestq) [inline datasets/wikicorpus/wikicorpus.py:97]: could you rename this column `tokens` ?

File: datasets/wikicorpus/wikicorpus.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+                    "lemmas": datasets.Sequence(datasets.Value("string")),
+                    "pos_tags": datasets.Sequence(datasets.Value("string")),
```
Comment (lhoestq) [inline datasets/wikicorpus/wikicorpus.py:99]: Could you use the `ClassLabel` feature type here ?

You can find an example of usage in the [flores](https://github.com/huggingface/datasets/blob/master/datasets/flores/flores.py) dataset

**Review by lhoestq — COMMENTED** [review]
Thanks the dummy_data are all good now!

I left comments about the feature types for POS tagging for consistency with other POS datasets.
Before we merge, can you  do this change please ?
You'll need to regenerate the dataset_infos.json after this change

File: datasets/wikicorpus/wikicorpus.py (hunk 4 above)
Comment (lhoestq) [reply datasets/wikicorpus/wikicorpus.py:99]: edit : conll, not flores sorry

Comment (albertvillanova) [reply datasets/wikicorpus/wikicorpus.py:99]: @lhoestq pos_tags in this dataset are not the same as the ones in conll. Here there are **hundreds** (maybe one thousand) of different PoS tag values, like: 
```
'DA0CS0', 'DA0FP0', 'DA0FS0', 'DA0MP0', 'DA0MS0', 'DD0CP0', 'DD0CS0', 'DD0FP0', 'DD0FS0', 'DD0MP0', 'DD0MS0', 
'DE0FS0', 'DE0MP0', 'DE0MS0', 'DI0CN0',...
...
'NC00000', 'NCCN000', 'NCCP000', 'NCCS000', 'NCFN000', 'NCFP000', 'NCFS000', 'NCMN000', 'NCMP000', 'NCMS000',
'NP00000',...
...
'P010P000', 'P010S000', 'P020P000', 'P020S000', 'P0300000', 'PD0CP000', 'PD0CS000', 'PD0FP000', 'PD0FS000', 
'PD0MP000', 'PD0MS000', 'PD0NS000', 'PI0CN000', 'PI0CP000',...
...
'VSIF2P0', 'VSIF2S0', 'VSIF3P0', 'VSIF3S0', 'VSII1P0', 'VSII1S0', 'VSII2P0', 'VSII2S0', 'VSII3P0', 'VSIP1P0', 
'VSIP1S0', 'VSIP2P0', 'VSIP2S0', 'VSIP3P0', 'VSIP3S0', 'VSIS1P0', 'VSIS1S0', 'VSIS2S0', 'VSIS3P0',...
```
Do you think it is a good idea to try to enumerate all of them?

Comment (albertvillanova) [reply datasets/wikicorpus/wikicorpus.py:99]: See more info here: https://freeling-user-manual.readthedocs.io/en/latest/modules/tagger/#relaxation-labelling-pos-tagger

Comment (lhoestq) [reply datasets/wikicorpus/wikicorpus.py:99]: Oh right I didn't know that.
Well then let's keep it this way then and let the users do what they want with the labels

File: datasets/wikicorpus/wikicorpus.py (hunk 3 above)
Comment (lhoestq) [reply datasets/wikicorpus/wikicorpus.py:97]: let's keep it this way actually with the raw format of the dataset as for the POS tags

**Review by lhoestq — APPROVED** [review]
All good then ! Thank you :)

**Pull request merged**